engine/esma_template.c

```diff
@@ -297,6 +297,19 @@ static int __decode_dbuf(char *line, struct esma_template *et)
 	s = p;
 	while (p != end) {
 
+		if ('/' == *p && '/' == *(p + 1)) {
+			while (*p != '\n')
+				p++;
+		}
+
+		if ('/' == *p && '*' == *(p + 1)) {
+			while (*p != '*' || *(p + 1) != '/') {
+				p++;
+			}
+			p += 2;	/* skiping '*' and '/' */
+			
+		}
```

Comment:
1. Unclosed `/*` may lead to off the buffer read due to the loop around `p++` without a check against `end`.
2. `p += 2` may overrun `end` as well, so we'll not only get off the buffer read, but, due to `p != end` condition (outer loop) will continue read off the buffer bounds.

Reply:
Thank you. Indeed, there was a buffer overflow in both types of comments. Fixed it and opened a PR: https://github.com/pcrew/esma/pull/16